src/utils/response.ts

```diff
@@ -220,7 +220,9 @@ export function sendStream(
   if (!stream || typeof stream !== "object") {
     throw new Error("[h3] Invalid stream provided.");
   }
-
+  if (Buffer.isBuffer(stream)) {
+    stream = Readable.from([stream]);
```

Comment:
I am wondering since the data source is already in memory as Buffer, would there be any benefits of this over simply using `res.end(stream)` and let Node.js/OS TCP stack handle streaming?

Reply:
Is there a guarantee it would send it as a stream?

Reply:
It is a TCP stream always. How to be chunked? it depends on many factors including stack and reverse proxy's behavior.

Reply:
@pi0 like this?